src/repository/opamVCS.ml

```diff
@@ -41,12 +41,12 @@ module Make (VCS: VCS) = struct
 
   let name = VCS.name
 
-  let fetch_repo_update repo_name ?cache_dir repo_root repo_url =
+  let fetch_repo_update repo_name ?full_fetch ?cache_dir repo_root repo_url =
     if VCS.exists repo_root then
       OpamProcess.Job.catch (fun e -> Done (OpamRepositoryBackend.Update_err e))
       @@ fun () ->
       OpamRepositoryBackend.job_text repo_name "sync"
-        (VCS.fetch ?cache_dir repo_root repo_url)
+        (VCS.fetch ?full_fetch ?cache_dir repo_root repo_url)
```

Comment:
@rjbou like this?
```suggestion
        (VCS.fetch ~full_fetch:false ?cache_dir repo_root repo_url)
```

Reply:
Linked to this https://github.com/ocaml/opam/pull/5888#discussion_r1548169887.

Reply:
not really. My comment was linked to a discussion during the dev meeting where we wondered if `?full_fetch` should be given explicitly or if it should be omitted when applied

Reply:
@kit-ty-kate did you want to keep `full_fetch:false` for `fetch_repo_update` that is used opam opam repo updates ?

Reply:
I would prefer personally but as you wish

Reply:
Good for me in that case :) It is a good safeguard regarding current opam repository.

Reply:
I'll add it